### ISSUE TITLE
Support access token retrieval from Moodle LMSes

### ISSUE THREAD

**Issue by pepeciavirella** [dmitry-viskov/pylti1.3#58]
Hey! I was just debugging through the `get_access_token` method of the `ServiceConnector` class and saw it expects successful (2xx or 3xx) responses to contain an `"access_token"` claim: https://github.com/dmitry-viskov/pylti1.3/blob/476556aa5b13da20be591b6aea4b4f61b85ea8fe/pylti1p3/service_connector.py#L87

I was recently working on an integration with a Moodle-based LMS, and realized that APIs for those kind of LMSes return a **claim with a different name** for the access token. Particularly, their OAuth Token URL's endpoint lives at the `/login/token.php`, and the token itself is provided as a `"token"` claim. e.g.:
```
{
  "token": "8b1cb0bebd237b919c71d7a8f487b4d2",
  "privatetoken": "U7XZRAaVmielUhkGcYVBKRxXEGhQ9wWCOeHQ2MyMR9jb6oA9TqZ2YdCKhJ7UgxGE"
}
```

The Moodle-sided code that returns this claim is defined here: https://github.com/moodle/moodle/blob/5ee0b620ae99bebeca9ee2c18078ee010913ae65/login/token.php#L104. Hence, I _think_ that a `KeyError` would be raised in the case where a Moodle API is hit to retrieve an access token, since the `"access_token"` claim would not be present even though the response is successful.

If you think this is case is worth considering, I would be happy to help in implementing it.

Best regards,
Pepe

**Comment by dmitry-viskov** [comment]
hi @pepeciavirella 
This looks odd because `access_token` is default naming. Have you already faced with the `KeyError` ? In my case I haven't ever faced with troubles in the integration with Moodle using `pylti1p3`

**Comment by pepeciavirella** [comment]
Hi!

Not _actually_, I haven't faced it at Moodle, but I've seen logs of an app I work on that experienced the `KeyError` on a Canvas integration (which is even more weird since its spec denotes an `"access_token"` return claim here); but I'd consider that an isolated case, it happened only once and it was a month ago.

Let me try triggering this retrieval at a Moodle integration once again, and see if I face this error or whatnot. I'll come back to you with results, in that case.

Best regards.

**Comment by pepeciavirella** [comment]
Hey @dmitry-viskov, I did some testing and I can state that `ServiceConnector`'s `get_access_token` **does work correctly** for Moodle instances. To sum it up, somehow I got the `/login/token.php` endpoint as OAuth Token URL configuration for the interface with one of those Moodle instances, but that endpoint has a different purpose: it's used to get access tokens **for Moodle's REST API**.

The actual endpoint to get LTI access tokens is `/lti/token.php` which, effectively, returns an `"access_token"` claim. Hence, this cases would succeed as I experienced yesterday when testing it specifically.

I'm sorry for the noise, this was a mis-configuration on my side. Feel free to close and/or delete this issue if you feel it, since this is not an issue at all.

Best regards,
Pepe
